content/6_Advanced/DP_More.mdx

```diff
@@ -105,16 +114,100 @@ Queue with 2 stacks is used to remove a factor of $\mathcal{O}(\log N)$ in [USAC
 
 ## DP on Broken Profile
 
-<Resources>
+<!-- <Resources>
 	<Resource
 		source="cp-algo"
 		title="DP on Broken Profile"
 		url="dynamic_programming/profile-dynamics.html"
 	>
 		{' '}
 	</Resource>
-</Resources>
+</Resources> -->
 
-<IncompleteSection />
+Broken profile DP is a subset of bitmask DP. Problems falling under this category generally have the following properties:
+1. They're about filling a 2D grid.
+2. One of the dimensions is much smaller than the other.
+3. When filling the grid, each cell depends only on adjacent cells.
+4. The cells don't have many possible values (usually only 2).
+
+The third property is especially important, as it means that we can process the cells column-by-column (imagine a snake wrapping around the grid). We then only need to care about the rightmost processed cell in each row (hence the name "broken profile").
+
+The fourth property suggests that we should use a bitmask to represent that broken profile.
+
+<Warning>
+
+Note that the bitmask doesn't necessarily have to represent the state of the *cells*. In some problems (e.g. CEOI 2006 Connect), it can represent the state of the cell *borders* instead.
+
+</Warning>
+
+<FocusProblem problem={problems.brokenFocus} />
+
+### Tutorial
+
+Let $\texttt{dp}[i][j][mask]$ be the number of ways to tile the grid so that:
+- All cells from cell $(0, 0)$ to cell $(i, j - 1)$ are **covered**.
+- All cells from cell $(i + 1, j)$ to cell $(N - 1, M - 1)$ are **empty**.
+- $mask$ represents whether each of the remaining $N$ cells are **empty**, with the $k$-th bit corresponding to the cell in column $k$.
+
+For example, the following state would contribute toward $\texttt{dp}[1][3][10100_2]$:
+![](broken-profile-state-example.png)
+
+The answer will be $\textt{dp}[N - 1][M - 1][0]$.
+
+We now have three cases when calculating $\texttt{dp}[i][j][mask]$:
+- The $j$-th bit of the mask is 0, meaning that cell $(i, j)$ is **covered**.
+	- Case 1: we used a vertical tile to cover it.
```

Comment:
now you've swapped horizontal / vertical?

Reply:
I realized that N = rows and M = columns in the CSES problem